### DISCUSSION TITLE
Misconfig? Help please

### DISCUSSION THREAD

**Discussion by nktine** [ExperienceLovelace/ha-floorplan#489] [Q&A]
Hello i discover this wonderfull card and tried to deal with it . Yet all my icon when cliqued do thé same action ):  the last state icon i enter . Here IS my code . IS there something i did wrong ( by thé way i didn't also manage to do direct action , the tap-action go to popup if anywome Can give advise too ..)
`type: picture-elements
elements:
  - type: image
    entity: light.soleil_switch_1
    style:
      top: 50%
      left: 50%
      width: 100%
    state_image:
      "on": /local/floor/switch.soleil_switch_1.png
      "off": /local/floor/transp.png
  - type: state-icon
    entity: light.soleil_switch_1
    style:
      top: 30%
      left: 45%
    icon: mdi:ceiling-light-multiple
    tap_action:
      action: perform-action
      perform_action: switch.toggle
      target:
        device_id: 2004aed44912c3994e86a0c25427d8bd
    hold_action:
      action: perform-action
      perform_action: light.toggle
      target:
        device_id: 2004aed44912c3994e86a0c25427d8bd
  - type: image
    entity: switch.plan_travail_cuisine_switch_1
    style:
      top: 50%
      left: 50%
      width: 100%
    state_image:
      "on": /local/floor/switch.plan_travail_cuisine_switch_1.png
      "off": /local/floor/transp.png
  - type: state-icon
    icon: mdi:light-flood-down
    entity: switch.plan_travail_cuisine_switch_1
    style:
      top: 35%
      left: 20%
    tap-action:
      action: toggle
  - type: image
    entity: switch.plafond_cuisine_switch_1
    style:
      top: 50%
      left: 50%
      width: 100%
    state_image:
      "on": /local/floor/switch.plafond_cuisine_switch_1.png
      "off": /local/floor/transp.png
  - type: state-icon
    entity: switch.plafond_cuisine_switch_1
    style:
      top: 40%
      left: 30%
    tap_action:
      action: toggle
    icon: mdi:lightbulb-alert-outline
    title: Plafond
  - type: image
    image: /local/floor/cover.store_cuisine_rideau.png
    style:
      left: 50%
      top: 50%
      width: 100%
    state_image:
      "on": /local/floor/cover.store_cuisine_rideau.png
      "off": /local/floor/transp.png
    entity: cover.store_cuisine_rideau
  - type: state-icon
    style:
      left: 34%
      top: 14%
    entity: cover.store_cuisine_rideau
    icon: mdi:blinds
    tap_action:
      action: perform-action
      perform_action: cover.toggle
      target:
        device_id: a4050ec9d1e08411351df180c849ff7b
  - type: image
    style:
      left: 50%
      top: 50%
      width: 100%
    image: /local/floor/cover.store_salon_rideau.png
    state_image:
      "on": /local/floor/cover.store_salon_rideau.png
      "off": /local/floor/transp.png
    entity: cover.store_salon_rideau
  - type: state-icon
    style:
      left: 49%
      top: 20%
    entity: cover.store_salon_rideau
    icon: mdi:blinds
    tap_action:
      action: perform-action
      perform_action: cover.toggle
      target:
        device_id: 408af0973689d70f99407a1e17979a6b
      data: {}
  - type: icon
    icon: mdi:cctv
    entity: camera.salon
    style:
      left: 47%
      top: 50%
image: /local/floor/eteint.png
`

**Comment by OzGav** [comment]
The yaml you have posted above is not correct at all for ha-floorplan. I am not sure where you got that from. Perhaps it is picture-elements?

**Reply by nktine** [reply]
Arg ... you're right i followed a old tutorial called floorplan  and didn't see it is from 2021

**Reply by exetico** [reply]
@nktine If ha-floorplan is something for you, check our Docs, and the videos here: https://youtube.com/playlist?list=PL5xKVw-BInX1phV-Tnjznwd2YG5mEOvWL&si=NU3T8AHP1SUW_I3s
